mingw-w64-clang/update-clang-from-msys2.sh

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/bash
+die () {
+	printf "$@" >&2
+	cleanup
+	exit 1
+}
+cleanup(){
+	rm -rf upstream
+}
+
+pkgname=mingw-w64-clang
+
+old_pkgver="$(sed -ne 's/pkgver=\([.0-9]*\).*/\1/p' -e 's/_version=\([.0-9]*\).*/\1/p' < PKGBUILD)"
+old_pkgrel="$(sed -ne 's/pkgrel=\([0-9]*\).*/\1/p' < PKGBUILD)"
+
+test -n "$old_pkgver" ||
+	die "$0: could not determine current pkgver\n"
+
+test -n "$old_pkgrel" ||
+	die "$0: could not determine current pkgrel\n"
+
+git clone --sparse --depth 1 https://github.com/msys2/MINGW-packages upstream
```

Comment:
How about `--filter=blob:none`? That makes the clone really fast, at the expense of adding a second (implicit) fetch in the next `git` invocation.